ansible/config_sonic_basedon_testbed.yml

```diff
@@ -291,6 +296,15 @@
 
       - debug: msg={{ docker_status.stdout_lines }}
 
+      - name: get platform name
+        shell: show platform summary | grep 'Platform' | awk '{print $2}'
+        register: platform
+
+      - name: rename platform.json before load_minigraph to let device use port_config.ini to generate port config
+        shell: mv /usr/share/sonic/device/{{ platform.stdout }}/platform.json /usr/share/sonic/device/{{ platform.stdout }}/platform.json.bak
```

Comment:
Could you elaborate the purpose of moving platform.json to platform.json.bak?

Reply:
This is unnecessary modification, will remove it

Reply:
> I don't understand why dynamically specifying vm_base is for concurrent deploy. If you need to concurrently deploy multiple topologies, then the VMs must not overlap. That means multiple test setups will co-exist on a same test server. Then, in testbed.csv, the vm_base must be correctly specified to avoid overlap in the first place. If vm_base is already correctly specified, why need to dynamically specify it again?

My headline is not suitable, it's not concurrency, this modifications should be related to the VMs' utilization. It depend on different deployment scenario of the testbed. If the content of testbed.csv is not dynamically/frequently generated , that means the VM base are defined solid. I would like to maximum the VM utilization in this kind of case. If there are any VMs are free, they can be dynamically assigned to new testing job, and be not restrict by the testbed.csv.

Reply:
Do you mean testbed.csv is outdated? The vm_base information in testbed.csv is too old and inaccurate?
We have to manually check if VMs are free, right? I am not sure whether it is easy to do that. Assume we have a large pool of VMs, it is usually not straightforward to tell which VMs are used by which testbed. Maybe the better way is to code correct vm_base info in testbed.csv.

Reply:
Maybe not outdated, in our testing environment, the vm_base of testbed.csv more likes the default value. There is extra service (e.g. Jenkins) to arrange the VM resources for testing. We also check the bridges' interfaces to get the VMs' usages, 
e.g. VM0212~VM0215 are allocated to the 2-7 testbed, the other VMs are free
```
sonic@sonic-server-02:~$ for i in $(sudo ovs-vsctl list-br | grep br-b); do echo ""; echo $i ; sudo ovs-vsctl list-ifaces $i ; done;
[sudo] password for sonic:

br-b-vms2-7
veth-bb-VM0212
veth-bb-VM0213
veth-bb-VM0214
veth-bb-VM0215
```
Therefore we add this extra argument to specify the VM base dynamically.